src/components/settings/SettingsAppearanceTab/CustomThemeModal.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, Button, Form, Hyperlink, Icon, ModalDialog,
+} from '@edx/paragon';
+import { InfoOutline } from '@edx/paragon/icons';
+import { HELP_CENTER_BRANDING_LINK } from '../data/constants';
+
+export default function CustomThemeModal({
+  isOpen, close, customColors, setTheme,
+}) {
+  const [button, setButton] = useState('');
+  const [buttonValid, setButtonValid] = useState(true);
+  const [banner, setBanner] = useState('');
+  const [bannerValid, setBannerValid] = useState(true);
+  const [accent, setAccent] = useState('');
+  const [accentValid, setAccentValid] = useState(true);
+
+  const headerText = 'Customize the admin and learner edX experience using your own brand colors. Enter color values in hexadecimal code.';
+  const infoText = 'More details about color appearance in the admin and learner experiences and best practices for selecting accessible colors are available in the ';
+  const invalidMessage = 'Must be hexidecimal starting with # (Ex: #1e0b57)';
+  const validHex = new RegExp('^#([A-Fa-f0-9]{6})$');
+
+  function validateColor(input, setter) {
+    if (!input.match(validHex)) {
+      setter(false);
+    } else {
+      setter(true);
+    }
+  }
+
+  function setCustom() {
+    const CUSTOM_THEME = {
+      title: 'Custom Theme',
+      button: (button === '' ? customColors?.button : button),
+      banner: (banner === '' ? customColors?.banner : banner),
+      accent: (accent === '' ? customColors?.accent : accent),
+    };
+    setTheme([CUSTOM_THEME, CUSTOM_THEME]);
+    close();
+  }
+
+  return (
+    <ModalDialog
+      title="My dialog"
+      isOpen={isOpen}
+      onClose={close}
+      size="md"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          Custom Theme
+        </ModalDialog.Title>
+        <p className="mt-3">{headerText}</p>
+      </ModalDialog.Header>
+
+      <ModalDialog.Body>
+        <Form>
+          <Form.Group controlId="custom-button">
+            <Form.Control
+              floatingLabel="Button color"
+              isInvalid={!buttonValid}
+              onChange={(e) => {
+                setButton(e.target.value);
+                validateColor(e.target.value, setButtonValid);
```

Comment:
[curious] While we don't have a true color picker component, it'd be pretty trivial to at least show a colored square or circle or something with the inputted hex value for each `Form.Control` as a quick preview for the admin to sanity check its indeed the correct color they intended.

Reply:
Maybe a follow-on task, just food for thought 😃

Reply:
I think this is a good idea, but I'll include it in my next ticket with adding accessibility checks to the custom modal :)

Reply:
love the idea!